engine/src/main/java/org/hibernate/validator/constraintvalidation/HibernateConstraintValidatorContext.java

```diff
@@ -68,4 +68,6 @@
 	 * @since 5.2
 	 */
 	TimeProvider getTimeProvider();
+
+	HibernateConstraintValidatorContext withInfo(Object info);
```

Comment:
withInfo was just something to have a base to discuss on. i pretty much like your proposal. going to change that :+1: